addons/web_linkedin/models/web_linkedin.py

```diff
@@ -0,0 +1,428 @@
+
+        image = record.get('pictureUrl') and self.url2binary(record['pictureUrl']) or False
+        data_dict['image'] = image
+
+        phone_numbers = (record.get('phoneNumbers') or {}).get('values', [])
+        for phone in phone_numbers:
+            if phone.get('phoneType') == 'mobile':
+                data_dict['mobile'] = phone['phoneNumber']
+            else:
+                data_dict['phone'] = phone['phoneNumber']
+        return data_dict
+
+    def get_search_popup_data(self, offset=0, limit=5, **kw):
+        """
+            This method will return all needed data for LinkedIn Search Popup.
+            It returns companies(including search by universal name), people, current user data and it may return warnings if any
+        """
+        result_data = {'warnings': []}
+        companies = {}
+        people = {}
+        headers = {'Content-type': 'application/json', 'Accept': 'text/plain', 'x-li-format': 'json'}
+        params = {
+            'oauth2_access_token': self.env.user.linkedin_token
+        }
+
+        #Profile Information of current user
+        profile_uri = "/people/~:(first-name,last-name)"
+        status, res = self.with_context(kw.get('local_context') or {}).send_request(profile_uri, params=params, headers=headers, type="GET")
+        result_data['current_profile'] = res
+
+        status, companies, warnings = self.get_company_data(offset, limit, params=params, headers=headers, **kw)
+        result_data['companies'] = companies
+        result_data['warnings'] += warnings
+        status, people, warnings = self.get_people_data(offset, limit, params=params, headers=headers, **kw)
+        if status:
+            result_data['people_status'] = status
+        result_data['people'] = people
+        result_data['warnings'] += warnings
+        return result_data
+
+    @api.model
+    def get_company_data(self, offset=0, limit=5, params={}, headers={}, **kw):
+        """This method will fetch company data for LinkedIn Company Search Popup
+            :param offset: no of company load after show more
+            :param limit: No of companies to load
+            :param params: send request Param data like Access token etc
+            :param headers: headers of requests
+            :param kw: arguments to controller to search
+            :returns: linkedin company search data, status if error and warnings
+        """
+        companies = {}
+        universal_company = {}
+        warnings = []
+        if not params:
+            params = {'oauth2_access_token': self.env.user.linkedin_token}
+        if not headers:
+            headers = {'Content-type': 'application/json', 'Accept': 'text/plain', 'x-li-format': 'json'}
+        #search by universal-name
+        if kw.get('search_uid'):
+            universal_search_uri = "/companies/universal-name={company_name}:{company_fields}".format(company_name=kw['search_uid'], company_fields='(' + ','.join(COMPANY_FIELDS) + ')')
```

Comment:
use urlparse to compose urls parameters (I don't think your code will work with companies special characters in the name)